### PR TITLE
Disable autocast in GroupNormalization for mixed precision stability

### LINKED ISSUE #22586: [Bug] Mixed Precision (float16) numerical instability in GroupNormalization with small epsilon
## Description

While using keras.mixed_precision.set_global_policy("mixed_float16"), the GroupNormalization layer produces NaN outputs when the epsilon parameter is set to a value smaller than the precision limit of float16 (e.g., $1e-12$). Since the smallest positive normalized value in FP16 is approximately $6.1e-5$, a very small epsilon effectively acts as zero during variance normalization, leading to numerical collapse.

## Poc

```
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
import keras
import numpy as np

keras.mixed_precision.set_global_policy("mixed_float16")
inputs = keras.Input(shape=(16, 16, 64))
# Epsilon 1e-12 causes NaN in FP16
x = keras.layers.GroupNormalization(groups=8, epsilon=1e-12)(inputs)
model = keras.Model(inputs, x)

extreme_data = np.random.uniform(low=60000, high=70000, size=(1, 16, 16, 64)).astype("float32")
output = model(extreme_data)
print(f"NaN detected: {np.any(np.isnan(output))}")


```

## Observed Result

```
NaN detected: True

```

## Expected Behavior

The layer should either:

* Automatically clip the epsilon to a safe minimum (e.g., 1e-7) when the policy is float16.

* Perform the normalization internal math in float32 regardless of the policy (Upcasting).

## Actual Behavior

The layer produces NaN values immediately when exposed to large input values or small epsilons in mixed_float16 mode.

## Notes
* This issue arises from float16 precision limits rather than user misuse.
* Other normalization layers (e.g., BatchNormalization) internally upcast to float32 to avoid similar issues.

### PR DESCRIPTION
**Pull request by Lawson-Darrow**
## Description
GroupNormalization is numerically unstable under mixed precision because it does
not disable autocast the way LayerNormalization does. Under a `mixed_float16`
policy the layer's inputs and its `gamma`/`beta` reads get downcast to float16
before the normalization moments are computed. Inputs above the float16 max
(~65504) overflow to `inf` and produce NaN outputs and large affine weights
lose precision.

This sets `self.autocast = False` and passes `autocast=False` on the `gamma` and
`beta` weights matching LayerNormalization. The layer level flag keeps inputs in
their original precision through the moment computation. `autocast=False` on the
weights stops them being read as float16 inside an autocast scope (Keras can
still enter `AutocastScope(compute_dtype)` even with the layer-level flag off so
both pieces are needed to match LayerNormalization).

Added two regression tests:
- `test_mixed_precision_large_input_no_nan`: a large input under `mixed_float16`
  must not produce NaN (the reported symptom).
- `test_large_value_within_autocast_scope`: mirrors LayerNormalization's test,
  asserting the affine weights keep full precision inside an autocast scope.

Both tests fail on the current code and pass with this change. The full
`group_normalization_test.py` suite passes locally on the TensorFlow backend
(13 passed).

Fixes #22586

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/layers/normalization/group_normalization.py
Comment (hertschuh) [inline keras/src/layers/normalization/group_normalization.py]: Can you remove this? This is no longer correct as the `compute_dtype` is no longer in the picture.

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for addressing this!

File: keras/src/layers/normalization/group_normalization.py
Comment (Lawson-Darrow) [reply keras/src/layers/normalization/group_normalization.py]: Done!

**Review by hertschuh — APPROVED** [review]
Thanks!
